### PR TITLE
Msi 2507 add support of extension attributes to search request

### PR DESCRIPTION
**Pull request by swnsma**
### Description (*)
Add support of Extension Attributes for Search Request.
Please proceed after PR #2487

### Fixed Issues (if relevant)
1. https://github.com/magento/inventory/issues/2507
### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by novikor** [comment]
Just to it

**Comment by swnsma** [comment]
Removed module for the fix as it already come with the changes in the core.

**Comment by swnsma** [comment]
Here we go again: https://github.com/magento/magento2/issues/24626
After the fix has been done, someone did refactoring which leave the functionality in broken state.

**Comment by swnsma** [comment]
PR in core: https://github.com/magento/magento2/pull/24627

**Review by novikor — CHANGES_REQUESTED** [review]
1. https://github.com/magento/magento2/pull/24627 is already merged - no fix is needed there anymore;
2. Do we really need a separate module for the extension?

**Comment by swnsma** [comment]
Hi @novikor
These changes are still not in `magento/inventory`.
Separated module will simplify procedure of removing the redundant extension after the fix will be released.

File: composer.lock [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "_readme": [
         "This file locks the dependencies of your project to a known state",
-        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file",
```
Comment (novikor) [inline composer.lock]: Is that OK?

**Review by novikor — CHANGES_REQUESTED** [review]
Okay

**Review by novikor — APPROVED** [review]

**Pull request merged**